### ISSUE TITLE
nolearn.lasagne: Create usage and caveats docs

### ISSUE THREAD

**Issue by dnouri** [dnouri/nolearn#47]
The intention here is to write some usage documentation for `nolearn.lasagne`, and also mention a few of the caveats that have led to user confusion and tickets before.

Let's collect a list of issues that require a note in the docs here:

- [ ] #7: train_test_split does not work for Pandas Series with non Dense Index
- [ ] #13: Ability to Shuffle Data Before Each Epoch
- [ ] #26: Option to make the nets more deterministic

[anything else?]

**Comment by cbonnett** [comment]
Given that Lasagne had it's first release recently, maybe it;s time to make a small push for nolearn documentation ? I'de be up for helping out.

**Comment by dnouri** [comment]
Good call.  I'll try and start this effort over the next week or two.  Thanks for offering your help.
